chia/data_layer/data_store.py

```diff
@@ -485,7 +491,7 @@ async def insert_root_from_merkle_blob(
                     """,
                     (root_hash, zstd.compress(merkle_blob.blob), store_id),
                 )
-            if update_cache:
+            if update_cache and self.recent_merkle_blobs.get_capacity() > 0:
```

Comment:
I think we could use the same mechanism as in the previous method for consistency?